src/main/java/pl/edu/zut/mad/schedule/data/model/api/Lesson.kt

```diff
@@ -3,6 +3,7 @@ package pl.edu.zut.mad.schedule.data.model.api
 import io.realm.RealmObject
 
 internal open class Lesson(var subject: String = "", var courseType: String = "",
-                           var room: String = "", var teacher: Teacher? = null,
-                           var reservationStatus: String = "", var timeRange: TimeRange? = null)
-    : RealmObject()
+    var room: String = "", var teacher: Teacher? = null,
```

Comment:
All these need to be nullable?

Reply:
Unfortunately yes, this is needed for default constructor which is mandatory for realm database.